src/pip/_internal/req/req_tracker.py

```diff
@@ -9,30 +9,67 @@
 import logging
 import os
 
+import pip._vendor.contextlib2 as contextlib2
+
 from pip._internal.utils.temp_dir import TempDirectory
 from pip._internal.utils.typing import MYPY_CHECK_RUNNING
 
 if MYPY_CHECK_RUNNING:
     from types import TracebackType
-    from typing import Iterator, Optional, Set, Type
+    from typing import Dict, Iterator, Generator, Optional, Set, Type, Union
     from pip._internal.req.req_install import InstallRequirement
     from pip._internal.models.link import Link
 
 logger = logging.getLogger(__name__)
 
 
-class RequirementTracker(object):
+@contextlib.contextmanager
+def update_env_context_manager(**changes):
+    # type: (str) -> Generator[None, None, None]
+    target = os.environ
 
-    def __init__(self):
-        # type: () -> None
-        self._root = os.environ.get('PIP_REQ_TRACKER')
-        if self._root is None:
-            self._temp_dir = TempDirectory(delete=False, kind='req-tracker')
-            self._root = os.environ['PIP_REQ_TRACKER'] = self._temp_dir.path
-            logger.debug('Created requirements tracker %r', self._root)
+    # Save values from the target and change them.
+    non_existent_marker = object()
+    saved_values = {}  # type: Dict[str, Union[object, str]]
+    for name, new_value in changes.items():
+        if name in target:
+            saved_values[name] = target[name]
         else:
```

Comment:
```suggestion
        except KeyError:
```

Reply:
Note the suggestion doesn't actually remove a leading space, but it is rendered that way.